dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -532,6 +534,23 @@ public Workflow getWorkflow(@ApiParam(hidden = true) @Parameter(hidden = true, n
         return workflow;
     }
 
+    @GET
+    @Path("/{workflowId}/workflowVersions")
+    @UnitOfWork
+    @ApiOperation(nickname = "getWorkflowVersions", value = "Return all versions in an entry", authorizations = {
+            @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = WorkflowVersion.class, responseContainer = "List", notes = "This is one of the few endpoints that returns the user object with populated properties (minus the userProfiles property)")
+    @Operation(operationId = "getWorkflowVersions", description = "Return all versions in an entry", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public Set<WorkflowVersion> getWorkflowVersions(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user,
+            @ApiParam(value = "workflowID", required = true) @Parameter(name = "workflowId", description = "id of the worflow", required = true, in = ParameterIn.PATH) @PathParam("workflowId") Long workflowId) {
+        Workflow workflow = workflowDAO.findById(workflowId);
+        checkEntry(workflow);
+        checkCanRead(user, workflow);
+
+        List<WorkflowVersion> versions = this.workflowVersionDAO.getWorkflowVersionsByWorkflowId(workflow.getId(), VERSION_PAGINATION_LIMIT, 0);
```

Comment:
Nice to have, but can wait -- it would be nice if `VERSION_PAGINATION_LIMIT` and `0` were query parameters, but then we'd also need additional headers, so maybe best not to open that can of worms yet.

Also, I think Gary's code only limits the number of versions for the `"/path/workflow/{repository}/published"` endpoint? But not for My Workflows? Is this going to cause problems? Maybe we do need the query params to simulate existing behavior.

Or is the idea that the UI would call the other APIs using the INCLUDE parameter, rending this moot?

Reply:
Fair question and/or new ticket. 

We already have https://dockstore.org/api/containers/published?offset=20&limit=10&filter=&sortCol=stars&sortOrder=desc and https://dockstore.org/api/workflows/published?offset=20&limit=10&filter=&sortCol=stars&sortOrder=desc&services=false

Reply:
I didn't end up using this in the UI. It only gets used for tests here so I could have used a query instead. But maybe we want to use this later? And yeah that's the only place Gary's code is used